awx/api/generics.py

```diff
@@ -528,6 +530,8 @@ def get_queryset(self):
         self.check_parent_access(parent)
         qs = self.request.user.get_queryset(self.model).distinct()
         sublist_qs = self.get_sublist_queryset(parent)
+        if not self.filter_read_permission:
+            return sublist_qs.prefetch_related(*self.prefetched_items)
```

Comment:
There's an alternative method of doing this above in this file, which can avoid the contract change of adding the `prefetched_items` property to the view

https://github.com/ansible/awx/blob/4f6c853c3ef4f0b87788b55c8cd58f4b5248a4fe/awx/api/generics.py#L367-L372

It would also be good to move the `qs` definition to after this `return` because in this code path it's never used.

Reply:
I mostly understood your comments and am going to try something like this:

if not self.filter_read_permission:
            access_class = access_registry[self.model]
            if access_class.prefetch_related:
                return sublist_qs.prefetch_related(*access_class.prefetch_related)
            if access_class.select_related:
                return sublist_qs.select_related(*access_class.select_related)
       I reordered the returns as it seems that prefetch_related if available returns with less DB queries even if it is a little more processing on the api